lib/utils/class-to-invokable.ts

```diff
@@ -1,28 +1,23 @@
-/**
- * Utility type for a class which can be called in addion to being used as a constructor.
- */
-interface Invokeable<Args extends Array<any>, Instance> {
-  (...args: Args): Instance;
-  new (...args: Args): Instance;
-}
+type Callable<A extends abstract new () => any> = A extends new (
+  ...args: infer Args
+) => infer Instance
+  ? A & ((...args: Args) => Instance)
+  : never;
 
 /**
  * Wraps a constructor to not need the `new` keyword using a proxy.
  * Only used for data types.
  *
- * @param {ProxyConstructor} Class The class instance to wrap as invocable.
+ * @param {ProxyConstructor} constructor The class instance to wrap as invocable.
  * @returns {Proxy} Wrapped class instance.
  * @private
  */
-export function classToInvokable<Args extends Array<any>, Instance extends object>(
-  Class: new (...args: Args) => Instance,
-): Invokeable<Args, Instance> {
-  return new Proxy<Invokeable<Args, Instance>>(Class as any, {
-    apply(_target, _thisArg, args: Args) {
-      return new Class(...args);
-    },
-    construct(_target, args: Args) {
-      return new Class(...args);
-    },
+export function classToInvokable<
+  Class extends new (...args: Array<never>) => any
```

Comment:
This could also be `Array<any>` (and that's what I'd normally put).  I just did this since eslint recommended it and I was unsure if it'd work.  Will probably change it back to `Array<any>` since that's more intuitive IMO.